### PR TITLE
Context Tree View

### PR DESCRIPTION
**Pull request by udgover**
This PR adds support to render context as a tree view. The screenshot below highlights current implementation and changes:

Screenshot 2024-12-24 at 08 55 57

### CODE REVIEW

File: src/views/ObservableDetails.vue [hunk 1]
```diff
@@ -354,8 +353,97 @@ export default {
...
+                root.push(element)
+                treeify(element.children, value)
+              }
+              else {
+                var title = key + ": " + value
+                var element = {"id": id++, "title": title}
+                root.push(element)
+              }
+            }
+        }
```
Comment (tomchop) [inline src/views/ObservableDetails.vue:397]: This seems very similar to the function in `jsonToTreeView`, can we somehow extract them?

File: src/views/ObservableDetails.vue [hunk 2]
```diff
@@ -354,8 +353,97 @@ export default {
...
+      }
+      for (const ctx of this.observable.context) {
+        var element = {"id": id++, "title": ctx.source, "children": [], "color": "warning"}
+        items.push(element)
+          treeify(element.children, ctx)
+      }
+      return items
+    },
+
+    jsonToTreeView(input: any) {
```
Comment (tomchop) [inline src/views/ObservableDetails.vue:407]: In fact, it looks like `jsonToTreeView` is never used?

**Review by tomchop — CHANGES_REQUESTED** [review]

File: src/views/ObservableDetails.vue (hunk 2 above)
Comment (udgover) [reply src/views/ObservableDetails.vue]: Duplicate while testing... Removed in eb701fe58a01fbc00bb64348deadbb14d205b651

File: src/views/ObservableDetails.vue (hunk 1 above)
Comment (udgover) [reply src/views/ObservableDetails.vue:397]: done in eb701fe58a01fbc00bb64348deadbb14d205b651

**Review by tomchop — APPROVED** [review]

**Pull request merged**